src/session/content/chat_action_bar.rs

```diff
@@ -282,11 +287,183 @@ impl ChatActionBar {
+                                  permissions: BoxedChatPermissions,
+                                  status: BoxedChatMemberStatus,
+                                  is_channel: bool| {
+                            if is_channel {
+                                matches!(
+                                    status.0,
+                                    ChatMemberStatus::Creator(_)
+                                        | ChatMemberStatus::Administrator(_)
+                                )
+                            } else {
+                                match status.0 {
+                                    ChatMemberStatus::Restricted(data)
+                                        if !data.permissions.can_send_messages =>
+                                    {
+                                        data.permissions.can_send_messages
+                                    }
+                                    ChatMemberStatus::Left | ChatMemberStatus::Banned(_) => false,
+                                    _ => permissions.0.can_send_messages,
+                                }
+                            }
+                        }),
+                    )
+                    .upcast()
+                }
+                ChatType::BasicGroup(data) => {
+                    let status_expression =
+                        gtk::ConstantExpression::new(data).chain_property::<BasicGroup>("status");
+                    gtk::ClosureExpression::new::<bool, _, _>(
+                        &[permissions_expression.clone(), status_expression],
+                        closure!(|_: Chat,
+                                  permissions: BoxedChatPermissions,
+                                  status: BoxedChatMemberStatus| {
+                            match status.0 {
+                                ChatMemberStatus::Restricted(data)
+                                    if !data.permissions.can_send_messages =>
+                                {
+                                    data.permissions.can_send_messages
+                                }
+                                ChatMemberStatus::Left | ChatMemberStatus::Banned(_) => false,
+                                _ => permissions.0.can_send_messages,
+                            }
+                        }),
+                    )
+                    .upcast()
+                }
+            };
+
+            let message_bar_binding =
+                visible_expresison.bind(&*imp.scrolled_window, "visible", Some(chat));
+            let send_button_binding =
+                visible_expresison.bind(&*imp.send_message_button, "visible", Some(chat));
+            bindings.push(message_bar_binding);
+            bindings.push(send_button_binding);
+
+            // Handle whether or not restriction label should be shown
+            let visible_binding = gtk::ClosureExpression::new::<bool, _, _>(
+                &[chat_expression],
```

Comment:
Shouldn't the "permissions_expressions" used here instead? Else, if a permission update arrives, this will never be fired.
Also the style can be simplified.

```rust
permissions_expression.chain_closure::<bool>(closure!(|/* bla */| {/* bla */});
```